Library/Homebrew/cmd/update.rb

```diff
@@ -241,12 +241,12 @@ def pull!(options = {})
       return
     end
 
+    @initial_revision = read_current_revision
```

Comment:
I feel this is quite a big semantic change. Previously, you've got a report for the changes pulled into `master` (or whatever the upstream branch is). By moving this here, you'll get a diff between your current working branch and whatever commit happens to be `master` after the update. Since I'm the one who moved that to where it is now in #42074, I'm not exactly happy to see this.